src/context.js

```diff
@@ -1,7 +1,29 @@
+const cldeventsv02 = require('cloudevents-sdk/v02');
+const cldeventsv03 = require('cloudevents-sdk/v03');
+
 class Context {
   constructor(request, response) {
     this.request = request;
     this.response = response;
+
+    // Unmarshall Knative events request to cloudevent
+    if (('ce-type' in request.headers) && request.headers['ce-type'].startsWith('dev.knative')
+      && !(typeof request.body === 'undefined') && ('ce-specversion' in request.headers)) {
+
+      var unmarshaller = null;
+      if (request.headers['ce-specversion'] == '0.2') {
+        unmarshaller = new cldeventsv02.HTTPUnmarshaller();
+      } else if (request.headers['ce-specversion'] == '0.3') {
+        unmarshaller = new cldeventsv03.HTTPUnmarshaller();
+      }
+      unmarshaller.unmarshall(request.body, request.headers)
+        .then(cldevent => {
+          this.cloudevent = cldevent;
+        }).catch(err => {
+          console.error(err);
```

Comment:
Would it be useful to add an `error` property to the `Context` and set that here?

Reply:
Good idea, will add that.